packages/react-dom/src/__tests__/ReactDOMFiber-test.js

```diff
@@ -1114,137 +1000,155 @@ describe('ReactDOMFiber', () => {
+    });
+    assertLog(['A']);
+  });
 
-    if (__DEV__) {
-      expect(console.error).toHaveBeenCalledTimes(2);
-      expect(console.error.mock.calls[0][0]).toMatch(
-        'ReactDOM.render is no longer supported in React 18',
-      );
-      expect(console.error.mock.calls[1][0]).toMatch(
-        'ReactDOM.render is no longer supported in React 18',
+  it('should not crash encountering low-priority tree', async () => {
+    await act(() => {
+      root.render(
+        <div hidden={true}>
+          <div />
+        </div>,
       );
-    }
-  });
+    });
 
-  it('should not crash encountering low-priority tree', () => {
-    ReactDOM.render(
-      <div hidden={true}>
-        <div />
-      </div>,
-      container,
-    );
+    expect(container.innerHTML).toBe('<div hidden=""><div></div></div>');
   });
 
-  it('should not warn when rendering into an empty container', () => {
-    ReactDOM.render(<div>foo</div>, container);
+  it('should not warn when rendering into an empty container', async () => {
+    await act(() => {
+      root.render(<div>foo</div>);
+    });
     expect(container.innerHTML).toBe('<div>foo</div>');
-    ReactDOM.render(null, container);
+    await act(() => {
+      root.render(null);
+    });
     expect(container.innerHTML).toBe('');
-    ReactDOM.render(<div>bar</div>, container);
+    await act(() => {
+      root.render(<div>bar</div>);
+    });
     expect(container.innerHTML).toBe('<div>bar</div>');
   });
 
-  it('should warn when replacing a container which was manually updated outside of React', () => {
+  it('should warn when replacing a container which was manually updated outside of React', async () => {
     // when not messing with the DOM outside of React
-    ReactDOM.render(<div key="1">foo</div>, container);
-    ReactDOM.render(<div key="1">bar</div>, container);
+    await act(() => {
+      root.render(<div key="1">foo</div>);
+    });
+    expect(container.innerHTML).toBe('<div>foo</div>');
+
+    await act(() => {
+      root.render(<div key="1">bar</div>);
+    });
     expect(container.innerHTML).toBe('<div>bar</div>');
+
     // then we mess with the DOM before an update
     // we know this will error - that is expected right now
     // It's an error of type 'NotFoundError' with no message
     container.innerHTML = '<div>MEOW.</div>';
 
     expect(() => {
-      expect(() =>
-        ReactDOM.render(<div key="2">baz</div>, container),
-      ).toErrorDev(
-        'render(...): ' +
-          'It looks like the React-rendered content of this container was ' +
-          'removed without using React. This is not supported and will ' +
-          'cause errors. Instead, call ReactDOM.unmountComponentAtNode ' +
-          'to empty a container.',
-        {withoutStack: true},
-      );
-    }).toThrowError();
+      ReactDOM.flushSync(() => {
+        root.render(<div key="2">baz</div>);
+      });
+    }).toThrow('The node to be removed is not a child of this node');
```

Comment:
Note: the old `console.error` only existed in the legacy root, the new root only throws.